### PR TITLE
#566: complete lb iterations support

### LINKED ISSUE #566: Support LB iterations
This is related to [vt#2375](https://github.com/DARMA-tasking/vt/issues/2375)

@lifflander if you want to add something

In the JSON for a phase:

```json
"phases":
  [
     {
       "id": ...,
       "tasks": ...,
       "communications": ...,
       "lb_iterations": [
         {
           "id": ...,
           "tasks": ...,
           "communications": ...
         }
       ]
     }
  ]
```

### PR DESCRIPTION
**Pull request by ppebay**
Fixes #566

### CODE REVIEW

**Comment by ppebay** [comment]
Implementation is complete @lifflander ; it needs more testing. However a first review would be welcome at this point.

Also I think that, re: initial design, replication the `"communications"` values is sub-optimal, as these are invariant across load-balancing iterations **within a given phase**.

File: src/lbaf/Execution/lbsRuntime.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def get_work_model(self):
         """Return runtime work model."""
         return self.__work_model
 
-    def execute(self, p_id: int, phase_increment: int=0):
+    def execute(self, p_id: int, phase_increment: int=0, lb_iterations=False):
```
Comment (cwschilly) [inline src/lbaf/Execution/lbsRuntime.py:124]: Flagging that this argument is not used in the function--is it necessary?

Comment (ppebay) [reply src/lbaf/Execution/lbsRuntime.py:124]: Ah yes, but the plan is to use it in the future after discussion about the functionality with @lifflander 
Thanks for noticing.

**Comment by ppebay** [comment]
@lifflander following discussion, changes pushed. Ready to be reviewed for good. Tx.

**Comment by pierrepebay** [comment]
Output for `synthetic-blocks.yaml`:

| synthetic-dataset-blocks0 | synthetic-dataset-blocks1 |
|:---:|:---:|
| synthetic-dataset-blocks2 | synthetic-dataset-blocks3 |
| synthetic-dataset-blocks4 | synthetic-dataset-blocks5 |
| synthetic-dataset-blocks6 | synthetic-dataset-blocks7 |
| synthetic-dataset-blocks8 | synthetic-dataset-blocks9 |

**Review by lifflander — APPROVED** [review]
This looks good once we get the communication issue sorted.

**Comment by ppebay** [comment]
@lifflander connectivity appears correct as per images generated on Linux by @pierrepebay 

**Phase 0 (initial):**
synthetic-dataset-blocks0_720

then **after 8 iterations (last iteration):**
synthetic-dataset-blocks8_720

which **has now the same connectivity as the next phase (1):**
synthetic-dataset-blocks9_720

**Comment by ppebay** [comment]
For reference (**N.B.:** `1-->4` and `4-->1` result in a single visible straight edge, whence 7 distinguishable edges in the images above vs. 8 below):

conf-blocks_720

**Review by lifflander — APPROVED** [review]
This looks good to me!

**Pull request merged**